src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -678,17 +668,17 @@ private String deleteRequest(String path) throws IOException {
     }
 
     @Override
-    public Iterable<SCMFile> getDirectoryContent(BitbucketSCMFile parent) throws IOException, InterruptedException {
+    public Iterable<SCMFile> getDirectoryContent(BitbucketSCMFile directory) throws IOException, InterruptedException {
         List<SCMFile> files = new ArrayList<>();
-        String path = encodePath(parent.getPath());
-        String ref = Util.rawEncode(parent.getRef());
+        String path = encodePath(directory.getPath());
+        String ref = URLEncoder.encode(directory.getRef(), "UTF-8");
         int start=0;
-        String url = String.format(API_REPOSITORY_PATH+"/browse/%s?at=%s", owner, repositoryName, path, ref);
+        String url = String.format(API_BROWSE_PATH, getUserCentricOwner(), repositoryName, path, ref);
```

Comment:
we have an actual constant for this exact URL, let's use it... also the known working code showed that we were not using the correct owner and this code would fail in the case of user centric owners